app/controllers/courses/entry_requirements_controller.rb

```diff
@@ -23,6 +27,17 @@ def update
 
   private
 
+    def set_missing_subject_errors
+      errors = {}
+      course.gcse_subjects_required.each do |subject|
+        if !params.dig(:course, subject)
+          errors[subject] = ["Pick an option for #{subject.titleize}"]
+        end
+      end
+
+      errors
+    end
```

Comment:
This logic can be simplified using `select` + `map`; we filter away all the subjects which do exist in the params, then each subject becomes a tuple of `[subject, "msg"]`. An array of these tuples becomes the hash we want with `.to_h`:

```ruby
def missing_subject_errors
  course.gcse_subjects_required
    .select { |subject| !params.dig(:course, subject) }
    .map { |subject| [subject, "Pick an option for #{subject.titleize}"] }
    .to_h
end
```

Reply:
Updated.
`[subject, "Pick an option for #{subject.titleize}"]` needed to be `[subject, ["Pick an option for #{subject.titleize}"]]`